holoviews/core/util.py

```diff
@@ -24,6 +26,57 @@
 except ImportError:
     dd = None
 
+
+
+
+class HashGenerator(json.JSONEncoder):
+    """
+    Extends JSONEncoder to generate a hashable string for as many types
+    of key as possible. These keys are supplied by widgets and Stream
+    parameters and may be nested. If an unrecognized object type is
```

Comment:
``HashableJSON`` might be better - the output is a JSON string with some customised support for additional types.